### ISSUE TITLE
Won't disconnect current user if connected to another one in another tab

### ISSUE THREAD

**Issue by ludojmj** [AxaGuilDEv/react-oidc#645]
#### Issue

"react-oidc-redux": "^3.1.7": Even if disconnected in another tab, the current session is still alive in the actual tab (until the token times out).
No such issue with "react-oidc-redux": "3.1.6".

#### Steps to Reproduce

**New tab 1**

- Link "protected1" or "protected2": connection to "**Alice**".

**New tab 2**

- https://demo.identityserver.io/Account/Logout
- https://demo.identityserver.io/Account/Login: connection to "**Bob**".

**Back to tab 1**

- Link "protected1" or "protected2": still being connected to "**Alice**".
- Won't switch to "**Bob**" session.

#### Expected

If disconnected in another tab, the current user must be disconnected in the working tab too.

#### Actual

Still connected to initial user even if another user has logged in in another tab.

### Versions

"@axa-fr/react-oidc-redux": "^3.1.7",

### Additional Details

**Installed packages:**

- "@axa-fr/react-oidc-redux": "^3.1.7",
- "oidc-client": "^1.11.5",
- "redux-oidc": "^3.1.7"

**Comment by guillaume-chervet** [comment]
Hi @ludojmj , do you have update something more than our lib?

**Comment by guillaume-chervet** [comment]
And also thank you for that awesomz bug report :)

**Comment by guillaume-chervet** [comment]
@ludojmj can you help for #611 issue ?
It is missing in the redux ans context readme

**Comment by guillaume-chervet** [comment]
Are you using inmemory storage?

**Comment by ludojmj** [comment]
I didn't update anything : I took the project as is.
Since I didn't update anything, yes I use InMemoryWebStorage.
Next step: trying without InMemoryWebStorage.

**Comment by ludojmj** [comment]
Same issue if I don't use InMemoryWebStorage.
"oidc-client": "^1.11.5",

**Comment by MEnenko** [comment]
Is there a solution to this problem? I have a similar situation. Help me solve it

**Comment by allemas** [comment]
I have the same problem here when I disconnect I'm always connected to another tab. 
I use InMemoryStorage.

"@axa-fr/react-oidc-context": "^3.1.7",

**Comment by lomboboo** [comment]
The same for me, is there any way to handle it?

**Comment by lomboboo** [comment]
Hello, thanks for quick response. I am using `v5.12.0`. My scenario is a follows (and this is really the only thing missing for me to cover all my cases):
- login with Tab1 in ApplicationA
- login with Tab2 in ApplicationB
- logout or delete `idsrv` cookie from ApplicationB
- in Tab1 I am still logged in, no sessiontermination events fired.
We previously used another library which actually handled this case, but your solution I like your solution more. Would be nice if there would be an option to establish "monitoring" OP somehow and give an ability to handle this case.

**Comment by guillaume-chervet** [comment]
Hi @lomboboo , thank you for your information.

Are you using serviceworker version?
When do you need it.

I think it is not too hard to implement, but i need some times.

**Comment by lomboboo** [comment]
We used redux-oidc which is based on oidc-client-js.
No, I don’t use service worker configuration, but can try if you think it should help.
How long do you think this implementation can take?

**Comment by guillaume-chervet** [comment]
Thank you for the information. I do not know about the time to do it. I have no idea how oidc client can do it because we are at client side. May be it use post message between tabs. I have to check how do oidc client do if you need app A oidc client be disconnected at the same time as application B oidc react. I need to use exactly the same mecanisum. Did you configure something special in oidc client to have this behavior? oidc react V3 was using oidc client but we dis not have the behavior.

**Comment by guillaume-chervet** [comment]
I think i understood  https://link.medium.com/OQ2r94gLirb
I will continue to investigate and test that.

**Comment by lomboboo** [comment]
I don't think it has anything to do with the tabs specifically. You can logout from a different browser and it still emits logout from chrome tab. Basically, when you logout your idsrv cookie is deleted from all browsers/tabs, maybe there is a watch on that. 
I can take a look how it is triggered on the weekend.

**Comment by lomboboo** [comment]
@guillaume-chervet Hello, I've spent some time looking of how it is implemented in the oidc-client-js. It is actually defined by [openid specs](https://openid.net/specs/openid-connect-session-1_0.html). Oidc-client-js implementation is actually very similar and handled by [CheckSessionIFrame](https://github.com/IdentityModel/oidc-client-js/blob/dev/src/CheckSessionIFrame.js) and [SessionMonitor](https://github.com/IdentityModel/oidc-client-js/blob/dev/src/SessionMonitor.js)

**Comment by guillaume-chervet** [comment]
Thank you very much for your help, i will try to implement that quickly.

**Comment by guillaume-chervet** [comment]
It is my next objective to set this behavior ul.

**Comment by lomboboo** [comment]
Great, looking forward to

**Comment by guillaume-chervet** [comment]
I was figthing with mozilla but it is just a cookie problem in local development :]

I have to finish the pr (finish behavior and lot of clean refactor), but I got it how it works. I think I will need few more days.
It is a great feature thank @lomboboo  for your help !

**Comment by guillaume-chervet** [comment]
Hi @lomboboo,

if you want you can make a try :

https://black-rock-0dc6b0d03-817.westeurope.1.azurestaticapps.net/

Published under version 6.0.0-alpha0
The breaking change is that it require a new silent_signin_uri property and the silent_redirect_uri need to be declared to OIDC server.

It seem stable, i need to clean and document it.

**Comment by guillaume-chervet** [comment]
here a quick migration guide https://github.com/AxaGuilDEv/react-oidc/blob/295776fbfeabde78f6d39ae5326f3a7da5050a11/MIGRATION_GUIDE_V5_TO_V6.md

**Comment by lomboboo** [comment]
@guillaume-chervet First of all thanks for the awesome feature!
I was wondering is it possible to pass `auth` or `oidc` object to the `sessionLostComponent`? For example, I would like to wait 5s and call `signinRedirect` there. I saw how you implemented this in the examples, but I think it would be much cleaner solution.
Also, can you please explain `silent_signin_uri` and `silent_redirect_uri` a bit more?

**Comment by guillaume-chervet** [comment]
Hi @lomboboo , if you have that screen it is that silentsignin won't work.

In my app i set up a button in order to signin again. I will send you a sample in a next message.

Does the logout work well for you?

**Comment by guillaume-chervet** [comment]
Here a sample used in production https://github.com/AxaGuilDEv/ml-cli/blob/master/src/Ml.Cli.WebApp/ClientApp/src/Server/shared/Oidc/SessionLost.component.js sample

**Comment by guillaume-chervet** [comment]
redirect_uri: manage callback inforation after login and must be declared to your OIDC server
silent_signin_uri: route that trigger silent signin

**Comment by lomboboo** [comment]
My scenario is that when I logout from different app or tab `sessionLost` is triggered and this is fine, cause there is no more session, but I want to be able to redirect user  with the `signinRedirect()`.

`redirect_uri` is specified - I was asking about `silent_redirect_uri`.  Should it be the same as `redirect_uri`?
`silent_signin_uri` is `check_session_iframe`? If so, it can be retrieved from the `.well-known/openid-configuration`

**Comment by guillaume-chervet** [comment]
exactly what does =>  https://github.com/AxaGuilDEv/ml-cli/blob/master/src/Ml.Cli.WebApp/ClientApp/src/Server/shared/Oidc/SessionLost.component.js

**Comment by guillaume-chervet** [comment]
I had to separe silent_redirect_uri and redirect_uri because the code differ to much

**Comment by lomboboo** [comment]
You didn't understand. Logout works as you described. When cookies are deleted `sessionLost` event is fired which redirects to the `SessionLostComponent` - and this is all fine. I just need to override SessionLostComponent with the logic that will call `signinRedirect`, so user is redirected to the login page. I saw how you implement it in your example with `useState` and rendering component inside the Context prodiver, but it seems unnecessary if you can pass `signinRedirect` function to the SessionLostComponent or even better the whole `auth/oidc` object.

`silent_signin_uri` is `check_session_iframe` from the [3.2 specs](https://openid.net/specs/openid-connect-session-1_0.html#toc)? If so, then it can be retrieved from the openid configuration endpoint

**Comment by guillaume-chervet** [comment]
I have added an onLogout event that you can inject inside OidcProvider.
It will take the control over sessionLostcomponent

If you set

function onLogout(){
   login()
}

It won't be triggered on manal logout() function.

I may need to rename it onLogoutFromAnotherTab.

**Comment by lomboboo** [comment]
where will the `login()` will come from? It is not passed anywhere in this `onLogout` function?

**Comment by guillaume-chervet** [comment]
Hi @lomboboo , login will come from useOidc hook. Check out that sample:
https://github.com/AxaGuilDEv/ml-cli/blob/master/src/Ml.Cli.WebApp/ClientApp/src/Server/shared/Oidc/SessionLost.component.js

In the beta 10 or before, i have renamed onLogout callback function
 to onLogoutFromAnotherTab.

With manual logout, it is a different use case, you have to create a callback route to give to you oidc server that will redirect to. You can do what ever you want on that route.

**Comment by lomboboo** [comment]
Can we then discuss it after your vacation?

**Comment by guillaume-chervet** [comment]
Hi @lomboboo , i am back from holiday. I am available.
